File: Resueltos_por_competidor/Juan_Martinez/Cakeminator.cpp
/*
 * Autor: EroMergeSort del TCC
 * Problema: Cakeminator 
 * Juez online: Vjudge
 * Veredicto: Accepted
 * Url: https://vjudge.net/problem/CodeForces-330A
 */

#include <bits/stdc++.h>
using namespace std;

int main() {
	int r, c, con = 0;
    char temp;
    cin >> r >> c;
    bool col[c], row[r];
    bool visited[r][c];

    
    for(int i = 0; i < r; i++){
        for(int j = 0; j < c; j++){
            visited[i][j] = false;
            //cout<< visited[i][j]<<" ";
        } 
           // cout<<"\n";
     }
    
    for(int i = 0; i < c; i++) col[i] = false;
     for(int i = 0; i < r; i++) row[i] = false;


    for(int i = 0; i < r; i++){
        for(int j = 0; j < c; j++){
            cin>>temp;
            if(temp == 'S') {
                col[j] = true;
                row[i] = true;
                //cout<< i << " " << j<<endl;
            }
        }
    }


    for(int i = 0; i < c; i++){
        //cout<<col[i]<<endl;
        if(!col[i]){
            for(int j = 0; j < r; j++){
                visited[j][i] = true;
            } 
         }
    }


    for(int i = 0; i < r; i++){
        if(!row[i]){
           for(int j = 0; j < c; j++){
            visited[i][j] = true;
        } 
        }
    }

    for(int i = 0; i < r; i++){
        for(int j = 0; j < c; j++){
            if(visited[i][j]) con++;
        } 
    }
    
     cout<<con<<endl;
}

File: Resueltos_por_competidor/Gabriella_Castro/Cakeminator.java
/*
 * Autor: Gabriella Castro
 * Problema: Cakeminator
 * Juez online: VJudge
 * Veredicto: Accepted
 * URL: https://vjudge.net/problem/CodeForces-330A
 */

import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] dimensiones = br.readLine().trim().split("\\s+");
        int r = Integer.parseInt(dimensiones[0]);
        int c = Integer.parseInt(dimensiones[1]);

        String[] mapa = new String[r];
        for (int i = 0; i < r; i++) {
            mapa[i] = br.readLine().trim();
        }
        int filasSeguras = 0;
        int columnasSeguras = 0;

        for (int i = 0; i < r; i++) {
            if (mapa[i].indexOf('S') == -1) {
                filasSeguras++;
            }
        }
        for (int j = 0; j < c; j++) {
            boolean fresa = false;
            for (int i = 0; i < r; i++) {
                if (mapa[i].charAt(j) == 'S') {
                    fresa = true;
                    break;
                }
            }
            if (!fresa) {
                columnasSeguras++;
            }
        }
        int comida = (filasSeguras * c) + (columnasSeguras * r) - (filasSeguras * columnasSeguras);
        System.out.println(comida);

    }

}
